dla_cnn/data_model/DesiMock.py

```diff
@@ -0,0 +1,88 @@
+
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
+        :param truth_path: truth file path
+        :param zbest_path: zbest file path
+        :return: self.wavelength,self.data(contained all information we need),self.split_point_br,self.split_point_rz,self.data_size
+        """
+        spec = fits.open(spec_path)
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        #spec[2].data ,spec[7].data and spec[12].data are the wavelength data
+        self.wavelength = np.hstack((spec[2].data.copy(), spec[7].data.copy(), spec[12].data.copy()))
+        self.data_size = len(self.wavelength)
+
+        dlas_data = truth[3].data[truth[3].data.copy()['NHI']>19.3]
+        spec_dlas = {}
+        #item[2] is the spec_id, item[3] is the dla_id, and item[0] is NHI, item[1] is z_qso
+        for item in dlas_data:
+            if item[2] not in spec_dlas:
+                spec_dlas[item[2]] = [('00'+str(item[3]-item[2]*1000),item[1],item[0])]
+            else:
+                spec_dlas[item[2]].append(('00'+str(item[3]-item[2]*1000),item[1],item[0]))
+
+        test = np.array([True if item in dlas_data['TARGETID'] else False for item in spec[1].data['TARGETID'].copy()])
+        for item in spec[1].data['TARGETID'].copy()[~test]:
+            spec_dlas[item] = []
+
+        spec_id = spec[1].data['TARGETID'].copy()
+        flux_b = spec[3].data.copy()
+        flux_r = spec[8].data.copy()
+        flux_z = spec[13].data.copy()
+        flux = np.hstack((flux_b,flux_r,flux_z))
+        ivar_b = spec[4].data.copy()
+        ivar_r = spec[9].data.copy()
+        ivar_z = spec[14].data.copy()
+        error = 1./np.sqrt(np.hstack((ivar_b,ivar_r,ivar_z)))
+        self.split_point_br = len(flux_b)
+        self.split_point_rz = len(flux_b)+len(flux_z)
+        z_qso = zbest[1].data['Z'].copy()
+        ra = spec[1].data['TARGET_RA'].copy()
+        dec = spec[1].data['TARGET_DEC'].copy()
+
+        self.data = {spec_id[i]:{'FLUX':flux[i],'ERROR': error[i], 'z_qso':z_qso[i] , 'RA': ra[i], 'DEC':dec[i], 'DLAS':spec_dlas[spec_id[i]]} for i in range(len(spec_id))}
+
+        return self.wavelength,self.data,self.split_point_br,self.split_point_rz,self.data_size
+
+    def get_sightline(self,id):
+        """
+        using id(int) as index to retrive each spectra in DesiMock's dataset, return  a Sightline object.
+        :param id: spectra's id , a unique number for each spectra
+        :return sightline:
+        """
+        sightline = Sightline(id)
+        sightline.flux = self.data[id]['FLUX']
+        sightline.error = self.data[id]['ERROR']
+        sightline.z_qso = self.data[id]['z_qso']
+        sightline.ra = self.data[id]['RA']
+        sightline.dec = self.data[id]['DEC']
+        sightline.dlas = self.data[id]['DLAS']

```

Comment:
This needs to be a list of `Dla` objects as defined in data_model/Dla.py

Reply:
Should I add those attributes to the Dla class since the Dla class only have entral_wavelength and col_density?

Reply:
yes, it is fine to add more attributes.

Reply:
Ok, I have updated it. Thanks for your advice!